WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -369,6 +369,9 @@ private extension MainTabBarController {
             return
         }
 
+        // Update view model with `siteID` to query correct Orders Status
+        viewModel.configureOrdersStatusesListener(for: siteID)
```

Comment:
Can you explain a little bit why this is needed here? Should we remove the setup on init?

Reply:
That's because `MainTabBarController` and its `MainTabViewModel` never deallocates. It is always the same instance when we switch stores (resulting different `siteID`s) and even users.